screen/features/src/main/java/com/ivy/features/FeaturesViewModel.kt

```diff
@@ -31,14 +31,26 @@ class FeaturesViewModel @Inject constructor(
     }
 
     @Composable
-    fun getFeatures(): ImmutableList<FeatureUi> {
-        val allFeatures = features.allFeatures.map {
-            FeatureUi(
-                name = it.name ?: it.key,
-                description = it.description,
-                enabled = it.asEnabledState()
-            )
+    fun getFeatures(): ImmutableList<Feature> {
+        val groups =
+            features.allFeatures.distinctBy { it.group }.map { it.group?.name ?: "" }.sorted()
+        val allFeatures: MutableList<Feature> = mutableListOf()
+        groups.forEach { group ->
```

Comment:
That's inefficient. Convert the features in a map using `associateBy { it.group }` and then for each group you can find its features in O(1) time. Now it's O(n) for each group

Reply:
Done